File: caesar.c
#include <cs50.h>
#include <math.h>
#include <string.h>
#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>
//make the user input arguments in the command line
int main(int argc, string argv[])
{
    //make sure the user did it right
    if (argc != 2)
    {
        printf("Usage <key>\n");
        return 1;
    }
    if (isalpha(argv[1]))
    {
        printf("Usage <key>\n");
        return 1;
    }
    //ask the user for the phrase to encrypt
    string plainText = get_string("plaintext: ");
    //change the argument to an int and declare variables
    int k = atoi(argv[1]);
    int num;
    int ciph;

    printf("ciphertext: ");
    //encryptor code
    for (int i = 0, l = strlen(plainText); i < l; i++)
    {
        //check if the character is alphabetical
        if (isalpha(plainText[i]))
        {
            if (isupper (plainText[i]))
            {
              num = plainText[i] - 65;
              num += k;
              ciph = num % 26;
              ciph += 65;
              printf("%c", (char)ciph);
            }
            else if (islower (plainText[i]))
            {
              num = plainText[i] - 97;
              num += k;
              ciph = num % 26;
              ciph += 97;
              printf("%c", (char)ciph);
            }
        }
        else
        {
            printf("%c", plainText[i]);
        }
    }
    //new line
    printf("\n");
}

File: caesar.py
from cs50 import get_string
import sys

if len(sys.argv) != 2:
    print("Exit Code 0: Incorrect number of arguments")

if sys.argv[1].isalpha():
    print("Exit Code 1: Contains numeric characters")
    sys.exit
plainText = get_string("plaintext: ")
k = int(sys.argv[1])
num = 0
ciph = 0
i = 0
for i in range(len(plainText)):
    if plainText[i].isalpha():
        if plainText[i].isupper():
            char = ord(plainText[i])
            num = char - 65
            num += k
            ciph = num % 26
            ciph += 65
            ciph = chr(ciph)
            print(f"{ciph}", end="")
        elif plainText[i].islower():
            char = ord(plainText[i])
            num = char - 97
            num += k
            ciph = num % 26
            ciph += 97
            ciph = chr(ciph)
            print(f"{ciph}", end="")
    else:
        print(f"{plainText[i]}", end="")
print("")